### PR TITLE
Add progress-level component

### PR DESCRIPTION
**Pull request by GabrielVidal1**
Added the progress-level component that will be used by the group manager list component to show different levels of permission.

### CODE REVIEW

File: src/app/modules/shared-components/components/progress-level/progress-level.component.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<div class="progress-level-container">
+  <ng-container *ngFor="let i of totalArray()">
```
Comment (smadbe) [inline src/app/modules/shared-components/components/progress-level/progress-level.component.html:2]: You cannot call a function from your html, otherwise angular call it each time there is a tiny unrelated change on the page. Add a console log in your function and you'll see.

File: src/app/modules/shared-components/components/progress-level/progress-level.component.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'alg-progress-level',
+  templateUrl: './progress-level.component.html',
+  styleUrls: ['./progress-level.component.scss']
+})
+export class ProgressLevelComponent {
+
+  @Input() size = 12;
```
Comment (smadbe) [inline src/app/modules/shared-components/components/progress-level/progress-level.component.ts:10]: not used? if not, remove

File: src/app/modules/shared-components/components/progress-level/progress-level.component.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  @Input() theme : 'default' | 'revert' = 'default';
+  @Input() filled = 0;
```
Comment (smadbe) [inline src/app/modules/shared-components/components/progress-level/progress-level.component.ts:12]: This attribute name is not clear... from what I understand it is just the number of dots filled... ok so it may just be called "value". And if you call "total" -> "maximum", you have the same kind of properties as a numeric slider... which is actually quite close to what this widget is.

Comment (GabrielVidal1) [reply src/app/modules/shared-components/components/progress-level/progress-level.component.ts]: fiixed

File: src/app/modules/shared-components/components/progress-level/progress-level.component.html (hunk 1 above)
Comment (GabrielVidal1) [reply src/app/modules/shared-components/components/progress-level/progress-level.component.html]: ok now I store the array in the component, and initialize it in the constructor.

File: src/app/modules/shared-components/components/progress-level/progress-level.component.ts (hunk 2 above)
Comment (GabrielVidal1) [reply src/app/modules/shared-components/components/progress-level/progress-level.component.ts]: fixed

File: src/app/modules/shared-components/components/progress-level/progress-level.component.ts [hunk 4]
```diff
@@ -9,13 +9,13 @@ export class ProgressLevelComponent {
 
   @Input() size = 12;
   @Input() theme : 'default' | 'revert' = 'default';
-  @Input() filled = 0;
-  @Input() total = 4;
+  @Input() value = 0;
+  @Input() maximum = 4;
 
-  constructor() {}
+  totalArray : number[] = [];
```
Comment (smadbe) [inline src/app/modules/shared-components/components/progress-level/progress-level.component.ts:14]: Detail: No need to put an initial value as you initial value in the constructor.
You should add explaining what this strange variable is.
```suggestion
  totalArray : number[]; // array containing [0, 1, 2, ...] as the template cannot iterate by itself
```

**Review by smadbe — CHANGES_REQUESTED** [review]

File: src/app/modules/shared-components/components/progress-level/progress-level.component.ts (hunk 4 above)
Comment (smadbe) [reply src/app/modules/shared-components/components/progress-level/progress-level.component.ts:14]: You just have "commit suggestion" directly in GH.

**Review by smadbe — APPROVED** [review]
